apps/omg/test/omg/state/core_test.exs

```diff
@@ -911,6 +913,227 @@ defmodule OMG.State.CoreTest do
+      {:ok, {block, _dbupdates}, _state} = form_block_check(state)
+
+      assert 2 = length(block.transactions)
+    end
+
+    test "should create utxos from claimed fee", %{alice: alice, state: state, fee_claimer: fee_claimer} do
+      state = Core.claim_fees(state)
+      {:ok, _, state} = form_block_check(state)
+
+      state
+      |> Core.exec(create_recovered([{1000, 1, 0, fee_claimer}], @eth, [{alice, 2}]), :no_fees_required)
+      |> success?()
+    end
+
+    test "fee txs cannot be intermixed with payments", %{alice: alice, state: state, fees: fees} do
+      fee_tx = create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, 2)
+
+      state
+      # fees from 1st tx are available to claim
+      |> Core.exec(fee_tx, fees)
+      |> success?()
+      # at this point no other payment can be processed
+      |> Core.exec(create_recovered([{1000, 0, 0, alice}], @eth, [{alice, 5}]), fees)
+      |> fail?(:payments_rejected_during_fee_claiming)
+    end
+
+    test "cannot claim the same token twice", %{state: state, fees: fees} do
+      fee_tx = create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, 2)
+
+      state
+      # fees from 1st tx are available to claim
+      |> Core.exec(fee_tx, fees)
+      |> success?()
+      # at this point no other payment can be processed
+      |> Core.exec(fee_tx, fees)
+      |> fail?(:surplus_in_token_not_collected)
+    end
+
+    test "cannot claim more than collected", %{state: state, fees: fees} do
+      paid_fee = 2
+
+      fee_tx = create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, paid_fee + 1)
+
+      state
+      |> Core.exec(fee_tx, fees)
+      |> fail?(:claimed_and_collected_amounts_mismatch)
+    end
+
+    test "cannot claim less than collected", %{state: state, fees: fees} do
+      paid_fee = 2
+
+      fee_tx = create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, paid_fee - 1)
+
+      state
+      |> Core.exec(fee_tx, fees)
+      |> fail?(:claimed_and_collected_amounts_mismatch)
+    end
+
+    test "no fees can be claimed after block is formed", %{state: state, fees: fees} do
+      fee_tx = create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, 2)
+
+      # now it's possible to claim Eth fee (note: no state modification)
+      state
+      |> Core.exec(fee_tx, fees)
+      |> success?()
+
+      # block is formed without claiming fees
+      {:ok, {_block, _dbupdates}, new_state} = form_block_check(state)
+
+      # it's no longer possible to claim fees
+      new_state
+      |> Core.exec(fee_tx, fees)
+      |> fail?(:surplus_in_token_not_collected)
+    end
+
+    test "fee is paid in one token only, many surpluses prohibited", %{alice: alice, state: state, fees: fees} do
+      state
+      |> do_deposit(alice, %{amount: 100, currency: @not_eth, blknum: 2})
+      |> Core.exec(
+        create_recovered([{1000, 0, 0, alice}, {2, 0, 0, alice}], [{alice, @eth, 5}, {alice, @not_eth, 90}]),
```

Comment:
`create_recovered` is a very bad name for what it really does. looking at this no one could know it's really creating transactions with inputs and ouputs

Reply:
Can't say - probably... it exists in the test-helpers since the beginning so I got familiar with it ;)